stats/stats.go

```diff
@@ -0,0 +1,688 @@
+	builder.WriteString(fmt.Sprintf("URL: %s\n", j.URL))
+	builder.WriteString(fmt.Sprintf("Status: %s\n", j.Status.Code))
+	builder.WriteString(fmt.Sprintf("Detailed Status: %s\n", j.Status.Message))
+	builder.WriteString(fmt.Sprintf("Local: %t\n", j.Local))
+	return builder.String()
+}
+
+type Status struct {
+	Code    string `json:"code"`
+	Message string `json:"message"`
+}
+
+type Service struct {
+	Type   string `json:"type"`
+	Status Status `json:"status"`
+}
+
+type License struct {
+	Type       string `json:"type"`
+	Expired    bool   `json:"expired"`
+	LicensedTo string `json:"licensed_to"`
+}
+
+type RepositoryDetails struct {
+	Key         string `json:"key"`
+	Type        string `json:"type"`
+	PackageType string `json:"packageType"`
+}
+
+type ReleaseBundleResponse struct {
+	ReleaseBundles []ReleaseBundleInfo `json:"release_bundles"`
+}
+
+type ReleaseBundleInfo struct {
+	RepositoryKey     string `json:"repository_key"`
+	ReleaseBundleName string `json:"release_bundle_name"`
+	ProjectKey        string `json:"project_key"`
+}
+
+func (rb ReleaseBundleInfo) String() string {
+	var builder strings.Builder
+
+	builder.WriteString(fmt.Sprintf("ReleaseBundleName: %s\n", rb.ReleaseBundleName))
+	builder.WriteString(fmt.Sprintf("RepositoryKey: %s\n", rb.RepositoryKey))
+	builder.WriteString(fmt.Sprintf("ProjectKey: %s\n", rb.ProjectKey))
+
+	return builder.String()
+}
+
+type GenericResultsWriter struct {
+	data   interface{}
+	format string
+}
+
+func NewGenericResultsWriter(data interface{}, format string) *GenericResultsWriter {
+	return &GenericResultsWriter{
+		data:   data,
+		format: format,
+	}
+}
+
+type StatsFunc func() (interface{}, error)
+
+func (sa *StatsArtifactory) GetCommandMap() map[string]StatsFunc {
+	return map[string]StatsFunc{
+		"rb":  sa.GetReleaseBundlesStats,
+		"jpd": sa.GetJPDsStats,
+		"rt":  sa.GetArtifactoryStats,
+		"pj":  sa.GetProjectsStats,
+	}
+}
+
+var processingOrders = []string{"pj", "rt", "jpd", "rb"}
+
+var printingOrders = []string{"rt", "pj", "jpd", "rb"}
+
+func (sa *StatsArtifactory) GetStats() error {
+
+	commandMap := sa.GetCommandMap()
+
+	allResultsMap := make(map[string]interface{})
+
+	filter := sa.FilterName
+
+	if filter != "" {
+		_, ok := commandMap[filter]
+		if !ok {
+			return fmt.Errorf("unknown filter: %s", filter)
+		}
+	}
+
+	if filter != "" {
+		allResultsMap[filter] = GetStatsUsingFilter(commandMap[filter])
+		if filter == "rt" {
+			allResultsMap["pj"] = GetStatsUsingFilter(commandMap["pj"])
+			updateProjectInArtifactoryInfo(&allResultsMap)
+			delete(allResultsMap, "pj")
+		}
+	} else {
+		for _, filter = range processingOrders {
+			allResultsMap[filter] = GetStatsUsingFilter(commandMap[filter])
+		}
+		updateProjectInArtifactoryInfo(&allResultsMap)
+	}
+	return sa.PrintAllResults(allResultsMap)
+}
+
+func (sa *StatsArtifactory) PrintAllResults(results map[string]interface{}) error {
+	for _, product := range printingOrders {
```

Comment:
you don't need to print based on product only if stats subcommand rt it should come to jfrog-cli-artifactory.